### PR TITLE
Add callbacks

### PR DESCRIPTION
**Pull request by leelew**
1. Add callbacks including early stopping, save best model in custom train;
2. Debug for custom training process (`@tf.function` cannot be call twice in `main` func);
3. Debug for LSTM (parameter `recurrent_dropout` in `tf.keras.layers.LSTM` must equal to 0, otherwise LSTM cannot be send to cuDNN accelerate, lead to much more time cost);
4. Change hidden size from 20 to 64, significantly increase the predictability.

### CODE REVIEW

**Comment by leelew** [comment]
5. Add static feat (maybe increase perform);
6. Add `num_repeat` params to perform exps with different random seeds;
7. Change `train_single` and `eval_single` to ensemble edition;
8. Separate `eval.py` from `train.py`;
9. Debug for optimizers, `Adam` is the best!
10. Change `valid_data` from tensor to nd.array to save GPU memory;
11. Red logs for validation;
12. Remove unreasonable runoff (>30mm/day);
13. Add `_spatial_normalize` for static feat;
14. RMSE loss;
15. Add dropout layer;
16. Refresh and re-train model when train NSE equal to NaN;
